smack-core/src/main/java/org/jivesoftware/smack/util/PacketParserUtils.java

```diff
@@ -558,7 +558,11 @@ public static Presence parsePresence(XmlPullParser parser)
                 case "show":
                     String modeText = parser.nextText();
                     if (StringUtils.isNotEmpty(modeText)) {
-                        presence.setMode(Presence.Mode.fromString(modeText));
+                        try {
+                            presence.setMode(Presence.Mode.fromString(modeText));
+                        } catch (IllegalArgumentException ex) {
+                            /* Invalid mode string received. */
```

Comment:
Please don't just silently swallow the exception. I would at least log it using `LOGGER` :)